### DISCUSSION TITLE
Pico W: hostname

### DISCUSSION THREAD

**Discussion by Batkin99** [micropython/micropython#10051] [General]
We're working on a pretty big project. The main part of the Micropython program for Rasperry pi pico W is already ready. We need for final testing the ability to dynamically change the dhcp hostname in STA_IF mode. Could you give an approximate timeline for update or at least get the beta version for testing?
Thanks

**Comment by jimmo** [comment]
This is a frequently requested feature! I'm not sure if this will make it for the upcoming v1.20 release, but hopefully soon after.

**Reply by Batkin99** [reply]
Thank you for your response. This feature is really necessary.

**Reply by billclawson** [reply]
I dug into the code a bit. I'm not sure exactly how the flags affect things, but the code for hostname is commented out in network_cyw43.c.  The hostname configuration lines are commented out using

#if !MICROPY_PY_NETWORK_CYW43_USE_LIB_DRIVER

MICROPY_PY_NETWORK_CYW43_USE_LIB_DRIVER gets set to 1 in CmakeLists.txt, and is contained in the code starting at line 245..
if (MICROPY_PY_NETWORK_CYW43)

That's as far as I took it.  Didn't want to start compiling the code as much of it is too unfamiliar to me.

**Reply by th3w** [reply]
There are currently two versions of the cyw43 driver in the Micropython source tree. For the Pico_W the driver is taken from /lib/cyw43-driver and the flag .MICROPY_PY_NETWORK_CYW43_USE_LIB_DRIVER=1 is set during compilation.

At the moment i found for the pico-w build at least the possibility to set a static default hostname via the macro CYW43_HOST_NAME when compiling the firmware. I added this in /port/rp2/boards/PICO_W/mpconfigport.h with the line  #define CYW43_HOST_NAME "uPY-PICOW".

See https://github.com/micropython/micropython/issues/10397#issuecomment-1372286591

Supplementary information. CYW43_HOST_NAME is used in /lib/cyw43-driver/src/cyw43_lwip.c:

[netif_set_hostname(n, CYW43_HOST_NAME);](https://github.com/georgerobotics/cyw43-driver/blob/2cf328d9e41603405a037a29e081a7d30dd519e6/src/cyw43_lwip.c#L205)

**Reply by jimmo** [reply]
What @th3w said is correct. The old driver supports hostname, but the new driver supports the rp2.

We're working on it. My in-progress branch has both ports using the new driver (lib/cyw43-driver), I'm just tidying up loose ends and sorting out things like the hostname API. (Just too many things on our plate)
